src/highdicom/io.py

```diff
@@ -242,31 +251,12 @@ def __init__(self, filename: Union[str, Path, DicomFileLike]):
         """
         if isinstance(filename, DicomFileLike):
             fp = filename
-            is_little_endian, is_implicit_VR = self._check_file_format(fp)
-            try:
-                if fp.is_little_endian != is_little_endian:
-                    raise ValueError(
-                        'Transfer syntax of file object has incorrect value '
-                        'for attribute "is_little_endian".'
-                    )
-            except AttributeError:
-                raise AttributeError(
-                    'Transfer syntax of file object does not have '
-                    'attribute "is_little_endian".'
-                )
-            try:
-                if fp.is_implicit_VR != is_implicit_VR:
-                    raise ValueError(
-                        'Transfer syntax of file object has incorrect value '
-                        'for attribute "is_implicit_VR".'
-                    )
-            except AttributeError:
-                raise AttributeError(
-                    'Transfer syntax of file object does not have '
-                    'attribute "is_implicit_VR".'
-                )
-            self._fp = fp
-            self._filename = Path(fp.name)
+            if isinstance(filename, DicomBytesIO):
+                self._fp = filename
+                self._filename = None
+            else:
+                self._fp = fp
+                self._filename = Path(fp.name)
```

Comment:
```suggestion
            self._fp = fp
            if isinstance(filename, DicomBytesIO):
                self._filename = None
            else:
                self._filename = Path(fp.name)
```

I suggest re-introducing the checks.